Source/com/drew/metadata/eps/EpsReader.java

```diff
@@ -0,0 +1,466 @@
+    {
+        try {
+            String[] tag = new String[]{"", ""};
+            tag[0] += (char)reader.getByte(pos);
+            pos++;
+
+            // Loop until data/binary starts OR available data has run out (not accurate, but should not be an issue as we are only interested in the beginning
+            while (pos < reader.getLength() && tag[0].hashCode() != EpsDirectory.TAG_BEGIN_BINARY
+                && tag[0].hashCode() != EpsDirectory.TAG_BEGIN_DATA
+                && tag[0].hashCode() != EpsDirectory.TAG_AI9_END_PRIVATE_DATA)
+            {
+                // Check if the comment has been found in the tag map - if so, extract the data
+                if (EpsDirectory._tagNameMap.containsKey(tag[0].hashCode())) {
+                    if (tag[0].hashCode() == EpsDirectory.TAG_DSC_VERSION) {
+                        extractDscData(directory, tag);
+                    } else if (tag[0].hashCode() == EpsDirectory.TAG_IMAGE_DATA) {
+                        extractImageData(directory);
+                    } else if (tag[0].hashCode() == EpsDirectory.TAG_BEGIN_PHOTOSHOP) {
+                        extractPhotoshopData(metadata);
+                    } else if (tag[0].hashCode() == EpsDirectory.TAG_BEGIN_ICC) {
+                        extractIccData(metadata);
+                    } else if (tag[0].hashCode() == EpsDirectory.TAG_BEGIN_XML_PACKET) {
+                        extractXmpData(metadata);
+                    } else if (tag[0].hashCode() == EpsDirectory.TAG_CONTINUE_LINE) {
+                        extractContinuation(directory, previousTag);
+                    } else {
+                        extractData(directory, tag);
+                    }
+                    if (tag[0].hashCode() != EpsDirectory.TAG_CONTINUE_LINE)
+                        previousTag = tag[0].hashCode();
+                    tag = new String[]{"", ""};
+                    tag[0] = "" + (char) reader.getByte(pos);
+                    pos++;
+                } else {
+                    previousTag = Integer.MAX_VALUE;
+                }
+
+                // Comments should never contain spaces/returns as per the EPS conventions
+                switch (tag[0].charAt(tag[0].length() - 1)) {
+                    case 0xA:
+                    case 0xD:
+                    case 0x9:
+                    case 0x20:
+                        tag[0] = "" + (char) reader.getByte(pos);
+                        pos++;
+                        while (tag[0].charAt(0) != 0x25) {
+                            tag[0] = "" + (char) reader.getByte(pos);
+                            pos++;
+                        }
+                        break;
+                    default:
+                        tag[0] += (char) reader.getByte(pos);
+                        pos++;
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
Got it, thanks!  I will fix the issues you have mentioned along with some of the styling like you had changed in the clipping path PR.  It has been a little while since I have worked on this PR, so I will go back through it and see if there's a good way to phase out the use of hash code!